docs/use_cases/fine_grained_authorization/dynamic.md

```diff
@@ -0,0 +1,416 @@
+# Implementing a dynamic FGA model
+
+What if we wanted to allow not only the object graph, but also the access control
+model itself to change at runtime? For instance, consider building a generic FGA platform
+like OpenFGA, where users can define new object types, relationships, and rules.
+With the approach we've taken so far, such changes would require creating a new SQL program
+and re-computing all derived relationships from scratch. In a large system, this may cause
+service disruptions, which may not be acceptable, especially if these changes occur frequently.
+
+In this section we build an FGA implementation that supports incremental updates to the
+authorization model in addition to incremental updates to the object graph.
+The key idea is to model FGA rules as relational data instead of hard-coding them as SQL queries.
+
+The implementation described below is available as a [pre-packaged use case
+in the Feldera online sandbox](try.feldera.com/create/?name=dynamic-fga) as well
+as in your local Feldera installation.
+
+:::note
+
+The implementation in this section prioritizes simplicity and clarity over performance.
+While it is not the most optimized solution, it is intentionally designed to effectively illustrate
+the concept.
+
+:::
+
+## Modeling objects and relationships
+
+In the interests of clarity, we use strings instead of integers for unique object IDs, allowing
```

Comment:
We can probably have UUID as a primitive this week.

Reply:
doesn't matter for this example, I could just use BIGINT's here, but for readability using names like `user:emily` is better.